src/main.cpp

```diff
@@ -1074,8 +1111,35 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
         }
     }
 
+    // Reject Overwintered transactions which have expired
+    if (tx.fOverwintered &&
+        tx.nVersion == 3 &&
+        tx.nFormatBranchId != 0 &&
+        tx.nExpiryHeight != std::numeric_limits<uint32_t>::max())
```

Comment:
This should go after `CheckTransaction()`; I'd put it between the coinbase check and `IsStandardTx()` check. Then we can remove the `tx.nFormatBranchId != 0` conditional.